### ISSUE TITLE
Why second argument (predicate) in partition not get key as other similar method ?

### ISSUE THREAD

**Issue by theGABS** [lodash/lodash#3891]
https://lodash.com/docs/4.17.10#partition

A lot of method which work with collection have format like
`The predicate is invoked with three arguments: (value, index|key, collection).`
But partition is not, it have different format
`The predicate is invoked with one argument: (value).` 

Why?

**Comment by jdalton** [comment]
Hi @theGABS!

See https://github.com/lodash/lodash/issues/2238.

**Comment by theGABS** [comment]
Hi, thank you for reply. I am not fully understand why exactly you not added it. but if it was intentionally then OK. I was just worried, maybe it is a bug.

**Comment by vicrac** [comment]
@jdalton I followed the discussion about this in other issues and I still don't quite get your point - it would be _extremely_ useful to have access to index while partitioning an array, consider splitting it into two halves (actually it is part of merge sort algorithm):

`const [left, right] = _.partition(array, (e,i) => i < array.length / 2)`

I really don't understand these arguments about "consistency". I would say that actual implementation is limiting and doesn't follow the principle of least astonishment - I would expect it to take the same arguments as others methods, like `_.map` or `_.filter`, I had to actually refer it in lodash docs after my tests failed - and I was totally surprised it works this way. Is there any hope for this to change?
